### ISSUE TITLE
The type or namespace name 'CreateBuilder' does not exist in the namespace 'WebApplication'

### ISSUE THREAD

**Issue by HybridSolutions** [dotnet/aspnetcore#38727]
### Description

The issue is simple to replicate. In VS 20222, if you create a new .net 6 ASP.NET Core Web App project named WebApplication, you will get the compile error  **CS0234 The type or namespace name 'CreateBuilder' does not exist in the namespace 'WebApplication'**

### Reproduction Steps

create a new .net 6 web project named WebApplication and try to compile it.

### Expected behavior

Should compile with no errors. Code straight from Visual Studio 2022 template generator.

### Actual behavior

you will get the compile error  **CS0234 The type or namespace name 'CreateBuilder' does not exist in the namespace 'WebApplication'**

### Known Workarounds

Couldn't figure how to fix the project since renaming it was not enough. Deleted it and created it again with the name MyWebApplication. The error is fixed.

### Other information

Found the solution to this issue at [https://youtrack.jetbrains.com/issue/RIDER-69896](https://youtrack.jetbrains.com/issue/RIDER-69896)

**Comment by davidfowl** [comment]
The problem is your application name is WebApplication, so the IDE thinks it's referring to the namespace instead of the build in WebApplication class. Rename the application and it should work.

**Comment by HybridSolutions** [comment]
> The problem is your application name is WebApplication, so the IDE thinks it's referring to the namespace instead of the build in WebApplication class. Rename the application and it should work.

Yes that is what I found out. That's why I submitted this as a bug because it shouldn't happen. The template wizard should not allow this.

**Comment by davidfowl** [comment]
A couple of things:
- The original issue is related to JetBrains not Microsoft.
- I don't think that's a reasonable to block this in the template wizard, it would need to block ALOT more names.
- It's possible to work around it even if you run into it. It would be nicer if the compile told you about ambiguous symbols though

**Comment by Muhammad-Atia** [comment]
> The problem is your application name is WebApplication, so the IDE thinks it's referring to the namespace instead of the build in WebApplication class. Rename the application and it should work.

**you absolutely  right**

**Comment by mbucek** [comment]
The main problem is, that "WebApplication" is the default name, and that default name is not allowed. So I think also this shall be fixed. (IDE shall use another default name, which does not end in a compile error CS0234 The type or namespace name 'CreateBuilder' does not exist in the namespace 'WebApplication'

**Comment by davidfowl** [comment]
Definitely, file an issue on rider.
